linode_api4/objects/object_storage.py

```diff
@@ -1,4 +1,50 @@
-from linode_api4.objects import Base, DerivedBase, Image, Property, Region
+from linode_api4.errors import UnexpectedResponseError
+from linode_api4.objects import Base, DerivedBase, Property, Region
+
+
+class ObjectStorageBucket(DerivedBase):
+    """
+    A bucket where objects are stored in.
+    """
+
+    api_endpoint = "/object-storage/buckets/{cluster}/{label}"
+    parent_id_name = "cluster"
+    id_attribute = "label"
+
+    properties = {
+        "cluster": Property(),
+        "created": Property(is_datetime=True),
+        "hostname": Property(),
+        "label": Property(),
+        "objects": Property(),
+        "size": Property(),
+    }
+
+    @classmethod
+    def api_list(cls):
+        """
+        Override this method to return the correct URL that will produce
+        a list of JSON objects of this class' type - Object Storage Bucket.
+        """
+        return "/".join(cls.api_endpoint.split("/")[:-2])
+
+    @classmethod
+    def make_instance(cls, id, client, parent_id=None, json=None):
+        """
+        Override this method to pass in the parent_id from the _raw_json object
+        when it's available.
+        """
+        if json is None:
+            return None
+        if parent_id is None and json["cluster"]:
+            parent_id = json["cluster"]
+
+        if parent_id:
+            return Base.make(id, client, cls, parent_id=parent_id, json=json)
```

Comment:
```suggestion
            return super().make(id, client, cls, parent_id=parent_id, json=json)
```
I would suggest using `super()` just in case we switch the parent class, this would be more likely to work.